src/ferc_xbrl_extractor/cli.py

```diff
@@ -76,139 +74,93 @@ def parse_main():
         "-a",
         "--archive-path",
         default=None,
-        type=str,
+        type=Path,
         help="Specify path to taxonomy entry point within a zipfile archive. This is a relative path within the taxonomy. If specified, `taxonomy` must be set to point to the zipfile location on the local file system.",
     )
     parser.add_argument(
         "-m",
         "--metadata-path",
         default=None,
-        type=str,
+        type=Path,
         help="Specify path to output metadata extracted taxonomy. Metadata will not be extracted if no path is specified.",
     )
     parser.add_argument(
         "--loglevel",
         help="Set log level (valid arguments include DEBUG, INFO, WARNING, ERROR, CRITICAL)",
         default="INFO",
     )
-    parser.add_argument("--logfile", help="Path to logfile", default=None)
+    parser.add_argument("--logfile", help="Path to logfile", type=Path, default=None)
 
     return parser.parse_args()
 
 
-def instances_from_zip(instance_path: Path) -> list[InstanceBuilder]:
```

Comment:
Moved these to `instance.py` for now, though I'm not quite sure if that's the right spot. There's a good argument for `xbrl.py` I think.